runner/app/live/pipelines/comfyui.py

```diff
@@ -0,0 +1,78 @@
+import os
+import json
+import torch
+from PIL import Image
+import asyncio
+import numpy as np
+
+from .interface import Pipeline
+from comfystream.client import ComfyStreamClient
+
+COMFY_UI_WORKSPACE_ENV = "COMFY_UI_WORKSPACE"
+DEFAULT_WORKFLOW_JSON = '''
+{
+    "1": {
+        "inputs": {
+            "images": ["2", 0]
+        },
+        "class_type": "SaveTensor",
+        "_meta": {
+            "title": "SaveTensor"
+        }
+    },
+    "2": {
+        "inputs": {
+            "engine": "depth_anything_vitl14-fp16.engine",
+            "images": ["3", 0]
+        },
+        "class_type": "DepthAnythingTensorrt",
+        "_meta": {
+            "title": "Depth Anything Tensorrt"
+        }
+    },
+    "3": {
+        "inputs": {},
+        "class_type": "LoadTensor",
+        "_meta": {
+            "title": "LoadTensor"
+        }
+    }
+}
+'''
+
+
+class ComfyUI(Pipeline):
+  def __init__(self, **params):
+    super().__init__(**params)
+
+    comfy_ui_workspace = os.getenv(COMFY_UI_WORKSPACE_ENV)
+    self.client = ComfyStreamClient(cwd=comfy_ui_workspace)
+
+    params = {'prompt': json.loads(DEFAULT_WORKFLOW_JSON)}
+    self.update_params(**params)
+
+    # Comfy will cache nodes that only need to be run once (i.e. a node that loads model weights)
+    # We can run the prompt once before actual inputs come in to "warmup"
+    warmup_input = torch.randn(1, 512, 512, 3)
+    asyncio.get_event_loop().run_until_complete(self.client.queue_prompt(warmup_input))
```

Comment:
Do we need to warmup only on startup or anytime the params change? I'd consider moving this to either `update_params` or in `process_frame` with a similar `self.first_frame` var, like the streamdiffusion pipeline (soft preference for the latter to avoid heavy processing on a constructor).

Reply:
The "warmup" probably only needs to happen on startup to ensure that the one-time expensive operations, like loading model weights, in the workflow are run before the first real input arrives. If the param values for any of the nodes in the workflow changes, then those nodes would be re-executed the next time the workflow is run.

Re: avoiding heavy processing in a constructor - for reference, in the pipeline for the comfystream example server, I perform this "warmup" operation in a [separate function](https://github.com/yondonfu/comfystream/blob/34bea08e3cd16fc1e0e4cc566961df881450b1f2/server/pipeline.py#L14) that ultimately just executes the workflow under the hood.

Reply:
Yeah, I think that we should either extend the "interface" to also include `warm` there (as used by Yondon). Otherwise, I think it's ok to keep it in the constructor.